tests/components/yolink/test_init.py

```diff
@@ -0,0 +1,67 @@
+"""Tests for the yolink integration."""
+
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from homeassistant.components.yolink import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.usefixtures("setup_credentials")
+async def test_device_remove_devices(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    mock_auth_mgr: MagicMock,
+    mock_yolink_home: MagicMock,
+) -> None:
+    """Test we can only remove a device that no longer exists."""
+    with (
+        patch(
+            "homeassistant.components.yolink.api.ConfigEntryAuth",
+            return_value=mock_auth_mgr,
+        ),
+        patch(
+            "homeassistant.components.yolink.YoLinkHome",
+            return_value=mock_yolink_home,
+        ),
+    ):
+        mock_config_entry = MockConfigEntry(
+            entry_id="test",
+            domain=DOMAIN,
+            title="yolink",
+            data={
+                "auth_implementation": DOMAIN,
+                "token": {
+                    "refresh_token": "mock-refresh-token",
+                    "access_token": "mock-access-token",
+                    "type": "Bearer",
+                    "expires_in": 60,
+                    "scope": "create",
+                },
+            },
```

Comment:
the unique id is also set to `DOMAIN`, set let's have that here as well

(Which I think we should either migrate to make sure people can set up 2 accounts)